blockwork/config/api.py

```diff
@@ -255,3 +269,15 @@ def __init__(self, api: ConfigApi, node: yaml.Node) -> None:
                 ),
             )
         )
+
+
+class TransformApi:
+    def __init__(self, api: ConfigApi, transform: "Transform") -> None:
+        self.api = api.fork(transform=self)
+        self.transform = transform
+        self.id = f"{type(transform).__name__}-{id(transform)}"
```

Comment:
It was highlighted earlier that `id(...)` might not be the best thing for us to be using as it's kinda random (i.e. it's just based on how Python decides to construct/place objects) and isn't guaranteed to be collision free run-to-run. Perhaps we should change this to use a datestamp instead?

Reply:
Maintaining the existing behaviour in this change, I'll make that its own change